dashboard/src/components/UpgradeForm/UpgradeForm.tsx

```diff
@@ -163,102 +175,96 @@ function UpgradeForm({
     if (availablePackageDetail) {
       const deployedSuccess = await dispatch(
         actions.apps.upgradeApp(
-          cluster,
-          namespace,
+          {
+            context: { cluster: cluster, namespace: namespace },
+            identifier: releaseName,
+            plugin: pluginObj,
+          } as InstalledPackageReference,
           availablePackageDetail,
           repoNamespace,
-          releaseName,
           appValues,
           schema,
         ),
       );
       setIsDeploying(false);
       if (deployedSuccess) {
-        dispatch(push(url.app.apps.get(cluster, namespace, releaseName, pluginObj)));
+        dispatch(
+          push(
+            url.app.apps.get({
+              context: { cluster: cluster, namespace: namespace },
+              plugin: pluginObj,
+              identifier: releaseName,
+            } as AvailablePackageReference),
+          ),
+        );
       }
     }
   };
 
-  if (error && (versions.length === 0 || !availablePackageDetail)) {
-    return (
-      <>
-        <LoadingWrapper>
-          <Alert theme="danger">An error occurred: {error.message}</Alert>
-        </LoadingWrapper>
-      </>
-    );
-  }
-
-  if (error || selected?.error) {
-    return (
-      <Alert theme="danger">
-        An error occurred: {error ? error.message : selected.error?.message}
-      </Alert>
-    );
-  }
-
-  if (versions?.length === 0 || !availablePackageDetail) {
-    return (
-      <LoadingWrapper
-        className="margin-t-xxl"
-        loadingText={`Fetching ${packageId}...`}
-        loaded={false}
-      />
-    );
-  }
-
   /* eslint-disable jsx-a11y/label-has-associated-control */
   return (
     <section>
-      <LoadingWrapper loaded={!isFetching}>
-        <ChartHeader
-          releaseName={releaseName}
-          chartAttrs={availablePackageDetail}
-          versions={versions}
-          onSelect={selectVersion}
-          currentVersion={deployed.chartVersion?.version?.pkgVersion}
-          selectedVersion={pkgVersion}
-        />
-        {isDeploying && (
-          <h3 className="center" style={{ marginBottom: "1.2rem" }}>
-            The application is being upgraded, please wait...
-          </h3>
+      {isDeploying && (
+        <h3 className="center" style={{ marginBottom: "1.2rem" }}>
+          The application is being upgraded, please wait...
+        </h3>
+      )}
+      {!isFetching && error && <Alert theme="danger">An error occurred: {error?.message}</Alert>}
+      <LoadingWrapper
+        loaded={!isDeploying && !isFetching && versions?.length > 0 && !!availablePackageDetail}
+      >
+        {(!isFetching && versions?.length === 0) || !availablePackageDetail ? (
+          <></>
+        ) : (
+          <>
```

Comment:
Similar thing as already commented herein: avoiding wrong transient error messages due to an incomplete page load,